praw/__init__.py

```diff
@@ -402,26 +423,11 @@ def handle_redirect():
                 assert url != request.url
             return response
 
-        request = _prepare_request(self, url, params, data, auth, files,
-                                   method)
         timeout = self.config.timeout if timeout is None else timeout
-
-        # Prepare extra arguments
-        key_items = []
-        oauth = request.headers.get('Authorization', None)
-        for key_value in (params, data, request.cookies, auth, oauth):
-            if isinstance(key_value, dict):
-                key_items.append(tuple(key_value.items()))
-            elif isinstance(key_value, http_cookiejar.CookieJar):
-                key_items.append(tuple(key_value.get_dict().items()))
-            else:
-                key_items.append(key_value)
-        kwargs = {'_rate_domain': self.config.domain,
-                  '_rate_delay': int(self.config.api_request_delay),
-                  '_cache_ignore': bool(files) or raw_response,
-                  '_cache_timeout': int(self.config.cache_timeout)}
+        request, key_items, kwargs = build_key_items()
 
         remaining_attempts = 3 if retry_on_error else 1
+        remaining_refresh = True
```

Comment:
Can we name this something like `attempt_oauth_refresh`? Also could it be initialized to `bool(self.refresh_token)`? That might simplify line 441.